substrate/frame/scheduler/src/tests.rs

```diff
@@ -2285,9 +2285,18 @@ fn postponed_named_task_cannot_be_rescheduled() {
 
 		// Run to a very large block.
 		run_to_block(10);
+
 		// It was not executed.
 		assert!(logger::log().is_empty());
-		assert!(Preimage::is_requested(&hash));
+
+		// Preimage was not available
+		assert_eq!(
+			System::events().last().unwrap().event,
+			crate::Event::CallUnavailable { task: (4, 0), id: Some(name) }.into()
+		);
+
+		// So it should not be requested.
+		assert!(!Preimage::is_requested(&hash));
```

Comment:
Some lines below this it is requested again while it wasn't executed. This looks like some potential bug to me. However, I didn't check the logic yet.

Reply:
yeah, but we do call `Preimage::note(call.encode().into()`

Reply:
Calling note through the trait instead of the `note_bytes` function does request it indeed.

Reply:
changed to use the extrinsic.